src/main/kotlin/lotto/domain/dto/PurchaseLottoRequest.kt

```diff
@@ -0,0 +1,18 @@
+package lotto.domain.dto
+
+import lotto.domain.LottoStore
+import lotto.domain.ManualLottoCount
+import lotto.domain.ManualLottoTickets
+import lotto.domain.PurchaseAmount
+
+class PurchaseLottoRequest(
+    val purchaseAmount: PurchaseAmount,
+    val manualLottoCount: ManualLottoCount,
+    val manualLottoTickets: ManualLottoTickets,
+) {
+    private val maxManualLottoCount = purchaseAmount.amount / LottoStore.LOTTO_PRICE
+
+    init {
+        require(maxManualLottoCount >= manualLottoCount.count) { println("최대 ${maxManualLottoCount}장 구매 가능합니다.") }
+    }
```

Comment:
객체를 초기화 할 때만 사용하는 정보이기 때문에, init 구문의 지역 변수로 선언해주셔도 될 것 같아요.
```suggestion
    init {
        val maxManualLottoCount = purchaseAmount.amount / LottoStore.LOTTO_PRICE
        require(maxManualLottoCount >= manualLottoCount.count)
    }
```